### ISSUE TITLE
Channels last doesn't improve speed when using SyncBatchNorm

### ISSUE THREAD

**Issue by tstandley** [pytorch/pytorch#50549]
## 🐛 Bug

Training with channels_last improves training speed significantly when using DistributedDataParallel.

But many projects benefit from SyncBatchNorm which allows training with smaller batch sizes because batch statistics can be pooled from multiple GPUs. Sometimes this improvement in model prediction quality can be extremely important.

Unfortunately, SyncBatchNorm doesn't work well with channels_last. Using channels_last actually causes training epochs to take several percent longer when SyncBatchNorm is being applied.

## To Reproduce

I've noticed this performance issue across a number of projects, but a good example can be found here:

https://github.com/tstandley/imagenet_training

here we train resnet152 with and without channels last, but with SBN.

`CUDA_VISIBLE_DEVICES=1,0 python train_imagenet.py -a=resnet152 -t=/data2/imagenet_data/train/ -v=/data2/imagenet_data/val/ -b=256 -j=4 --fp16 -sbn` (sync batch norm, but no channels_last)
results in epochs that take about 59 minutes

`CUDA_VISIBLE_DEVICES=1,0 python train_imagenet.py -a=resnet152 -t=/data2/imagenet_data/train/ -v=/data2/imagenet_data/val/ -b=256 -j=4 --fp16 -sbn -cl` (sync batch norm, and channels_last)
results in epochs that take about 65 minutes

For what it's worth, using channels last and no sbn results in epochs that take 40 minutes, and using neither results in epochs that take 47 minutes.
## Expected behavior

Ideally, channels last memory format should improve the speed even when using SyncBatchNorm. Thanks!

## Environment
PyTorch version: 1.7.1
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.1 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: Could not collect

Python version: 3.8 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: TITAN RTX
GPU 1: TITAN RTX
GPU 2: TITAN RTX

Nvidia driver version: 460.32.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.2
[pip3] numpydoc==1.1.0
[pip3] torch==1.7.1
[pip3] torchaudio==0.7.0a0+a853dff
[pip3] torchvision==0.8.2
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               10.2.89              hfd86e86_1  
[conda] mkl                       2020.2                      256  
[conda] mkl-service               2.3.0            py38he904b0f_0  
[conda] mkl_fft                   1.2.0            py38h23d657b_0  
[conda] mkl_random                1.1.1            py38h0573a6f_0  
[conda] numpy                     1.19.2           py38h54aff64_0  
[conda] numpy-base                1.19.2           py38hfa32c7d_0  
[conda] numpydoc                  1.1.0              pyhd3eb1b0_1  
[conda] pytorch                   1.7.1           py3.8_cuda10.2.89_cudnn7.6.5_0    pytorch
[conda] torchaudio                0.7.2                      py38    pytorch
[conda] torchvision               0.8.2                py38_cu102    pytorch

**Comment by melgor** [comment]
This is exactly what I discovered today. 
Setup:
```
Collecting environment information...
PyTorch version: 1.7.1
Is debug build: False
CUDA used to build PyTorch: 11.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.10.2

Python version: 3.7 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration:
GPU 0: Tesla V100-SXM2-16GB
GPU 1: Tesla V100-SXM2-16GB

Nvidia driver version: 450.51.06
cuDNN version: Probably one of the following:
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn.so.8
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_adv_train.so.8
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_ops_train.so.8
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] efficientnet-pytorch==0.7.0
[pip3] numpy==1.19.3
[pip3] torch==1.7.1
[pip3] torch-dct==0.1.5
[pip3] torchaudio==0.7.0a0+a853dff
[pip3] torchelastic==0.2.0
[pip3] torchvision==0.2.2
[conda] blas                      1.0                         mkl
[conda] cudatoolkit               11.0.221             h6bb024c_0
[conda] efficientnet-pytorch      0.7.0                    pypi_0    pypi
[conda] mkl                       2020.2                      256
[conda] mkl-service               2.3.0            py37h8f50634_2    conda-forge
[conda] mkl_fft                   1.2.0            py37h161383b_1    conda-forge
[conda] mkl_random                1.2.0            py37h9fdb41a_1    conda-forge
[conda] numpy                     1.19.3                   pypi_0    pypi
[conda] numpy-base                1.19.2           py37hfa32c7d_0
[conda] pytorch                   1.7.1           py3.7_cuda11.0.221_cudnn8.0.5_0    pytorch
[conda] torch-dct                 0.1.5                    pypi_0    pypi
[conda] torchaudio                0.7.2                      py37    pytorch
[conda] torchelastic              0.2.0                    pypi_0    pypi
[conda] torchvision               0.2.2                      py_3    pytorch
```


My results from DDP on 2xV100. In general `Channel-last` faster without `Sync-BN` and slower when model use `Sync-BN`
```
**Channel-First:**
Mean iteration duration: 0.3026939654350281, std_dev: 0.0898759985575116


**Channel-last:**
Mean iteration duration: 0.3513329601287842, std_dev: 0.0922589948304029


**Channel-First no Sync-BN:**
Mean iteration duration: 0.2425852108001709, std_dev: 0.09982567172989691


**Channel-last no Sync-BN:**
Mean iteration duration: 0.22269668579101562, std_dev: 0.07577016268206072
```

**Comment by hukkai** [comment]
Just an unrelated question. "Training with channels_last improves training speed significantly when using DistributedDataParallel." Could you give any references about this? The speed difference seems significant, but I just know it for the first time LOL. Thanks!

**Comment by ngimel** [comment]
For syncbatchnorm #46906 should help, we should review and land it.

**Comment by myron** [comment]
same issue with 5D tensors (3D images) using SyncBatchNorm.  It's 20% slower in channels_last in AMP with DDP (using latest pytorch and cudnn)
